packages/shared/profiles/sagas.ts

```diff
@@ -183,52 +197,91 @@ export function* handleFetchProfile(action: ProfileRequestAction): any {
-      avatar.hasConnectedWeb3 = true
+  try {
+    const userIdsToFetch = userIds.filter((userId) => userId.toLowerCase() !== identity.address.toLowerCase())
+    const isFetchingOwnUser = userIds.some((userId) => userId.toLowerCase() === identity.address.toLowerCase())
+
+    const avatars: Avatar[] = yield (!isFetchingOwnUser && call(getRemoteProfiles, userIdsToFetch, version)) || []
 
-      return avatar
+    const ownProfile = isFetchingOwnUser && (yield call(readProfileFromLocalStorage))
+    if (ownProfile != null) {
+      avatars.push(ownProfile)
     }
+
+    yield put(profilesSuccess(avatars))
   } catch (error: any) {
-    if (error.message !== 'Profile not found') {
-      defaultLogger.log(`Error requesting profile for auth check ${userId}, `, error)
+    trackEvent('error', {
+      context: 'kernel#saga',
+      message: `Error requesting profiles for ${userIds}: ${error}`,
+      stack: error.stack || ''
+    })
+    yield put(profilesFailure(userIds, `${error}`))
+  }
+}
+
+function* getRemoteProfiles(userIds: string[], version?: number) {
+  try {
+    const profiles: RemoteProfile[] = (yield call(profilesServerRequest, userIds, version)) as any
+
+    let avatars = profiles.map((profile) => {
+      let avatar = profile.avatars[0]
+
+      if (avatar) {
+        avatar = ensureAvatarCompatibilityFormat(avatar)
+        if (!validateAvatar(avatar)) {
+          defaultLogger.warn(`Remote avatar for users is invalid.`, userIds, avatar, validateAvatar.errors)
+          return null
+        }
+
+        avatar.hasClaimedName = !!avatar.name && avatar.hasClaimedName // old lambdas profiles don't have claimed names if they don't have the "name" property
+        avatar.hasConnectedWeb3 = true
+
+        return avatar
+      }
+    })
+
+    return avatars
+  } catch (error: any) {
+    if (error.message !== 'Profiles not found') {
+      defaultLogger.log(`Error requesting profiles for auth check ${userIds}, `, error)
     }
   }
   return null
 }
 
-export async function profileServerRequest(userId: string, version?: number) {
+export async function profilesServerRequest(userIds: string[], version?: number): Promise<RemoteProfile[]> {
   const state = store.getState()
   const catalystUrl = getCatalystServer(state)
 
   try {
-    let url = `${catalystUrl}/lambdas/profiles?id=${userId}`
+    let url = `${catalystUrl}/lambdas/profiles`
     if (version) url = url + `&version=${version}`
 
-    const response = await fetch(url)
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ ids: userIds })
+    })
 
     if (!response.ok) {
       throw new Error(`Invalid response from ${url}`)
     }
 
     const res = await response.json()
 
-    return res[0] || { avatars: [] }
+    return res || [{ avatars: [], timestamp: Date.now }]
```

Comment:
Shouldn't the multiple result be inside `avatars`? why do we have an array inside another array?

Reply:
That's because the response from the Backend responds a profile that contains an array of avatars, as seen in the new type `RemoteProfile` it's always been that way

Reply:
A profile can contain multiple avatars

Reply:
ok, thats a new one. do you know how does it work? because you can only deploy one profile per address

Reply:
I don't really know why the response is built that way, I'll sync with the explorer's and platform team to understand why the decision of multiple avatars per profile.

On another hand what I do know is that currently only the first avatar of each profile is being used as seen here https://github.com/decentraland/kernel/blob/79edd230d0c2b5cc43b7a6d7a1bcc9e4050b7d3a/packages/shared/profiles/sagas.ts#L438

Reply:
So after syncing with the platform team, apparently they just have an array BUT it's always a size 1 array, that's why everywhere you'll find `profile.avatars[0]` because there's no situation where avatars can have more than one. 
Still, changing the API would be a breaking change, and that's why it wasn't done (there are too many dependencies on that endpoint now for it to change)